### PR TITLE
feat: Add support for max number of tags allowed

### PR DESCRIPTION
**Pull request by faizanu94**
This PR adds support for max number of tags through maxTags (optional) prop, reference issue #920

### CODE REVIEW

File: __tests__/reactTags.test.js [hunk 1]
```diff
@@ -934,4 +934,22 @@ describe('Test ReactTags', () => {
...
+    });
+
+    it('should render input when tags are added less than maxTags value', () => {
+      const RenderResult = render(mockItem({ maxTags: 2 }));
+      expect(RenderResult.queryByTestId('input')).to.not.be.null;
+    });
+
+    it('should not render input when tags are added upto maxTags value', () => {
+      const RenderResult = render(mockItem({ maxTags: 1 }));
+      console.log('aa', RenderResult.queryByTestId('input'));
```
Comment (ad1992) [inline __tests__/reactTags.test.js:951]:
```suggestion
```

removing logs :)

File: __tests__/reactTags.test.js [hunk 2]
```diff
@@ -934,4 +934,22 @@ describe('Test ReactTags', () => {
       expect(onClearAllStub.calledOnce).to.be.true;
     });
   });
+
+  describe('When maxTags is defined', () => {
+    it('should render input when maxTags is nor defined', () => {
```
Comment (ad1992) [inline __tests__/reactTags.test.js:939]:
```suggestion
    it('should render input when maxTags is not defined', () => {
```

File: src/components/ReactTags.js [hunk 3]
```diff
@@ -501,27 +503,29 @@ class ReactTags extends Component {
...
-          onKeyDown={this.handleKeyDown}
-          onPaste={this.handlePaste}
-          name={inputName}
-          id={inputId}
-          maxLength={maxLength}
-          value={inputValue}
-          data-automation="input"
-          data-testid="input"
-        />
+        {(!maxTags || (maxTags && tags.length < maxTags)) && (
```
Comment (ad1992) [inline src/components/ReactTags.js:506]: lets return null if max tags reached instead for better readability

**Review by ad1992 — CHANGES_REQUESTED** [review]
Thank you for the PR @faizanu94, requested some changes :)

File: src/components/ReactTags.js (hunk 3 above)
Comment (faizanu94) [reply src/components/ReactTags.js]: thanks for the suggestion :)

File: src/components/ReactTags.js [hunk 4]
```diff
@@ -501,27 +503,29 @@ class ReactTags extends Component {
...
-          onKeyDown={this.handleKeyDown}
-          onPaste={this.handlePaste}
-          name={inputName}
-          id={inputId}
-          maxLength={maxLength}
-          value={inputValue}
-          data-automation="input"
-          data-testid="input"
-        />
+        {maxTags && tags.length >= maxTags ? null : (
```
Comment (ad1992) [inline src/components/ReactTags.js:506]: @faizanu94 lets do early return `null` since nothing will be rendered when max tags reached

**Review by ad1992 — CHANGES_REQUESTED** [review]

File: src/components/ReactTags.js (hunk 4 above)
Comment (faizanu94) [reply src/components/ReactTags.js:506]: @ad1992 it appears that an early return won't result in any rendering. Our objective here is to only make the `input` tag conditional. The rest of the JSX, including the Suggestions component, will be rendered regardless of this condition. Please correct me if I'm mistaken

Comment (ad1992) [reply src/components/ReactTags.js:506]: Got it, yes you are right,how about disabling the input instead of hiding it ? that could be better UI and we can show an message that max tags reached below the input as well

Comment (faizanu94) [reply src/components/ReactTags.js:506]: Agreed, this approach would enhance the UI. However, I'm just curious about the existing `disabled` attribute as defined in the `inputProps` of the library (https://github.com/react-tags/react-tags?tab=readme-ov-file#inputprops). There's a possibility that this could introduce edge cases that may conflict with our requirements (for example: when user explicitly passes `disabled={false}` prop). In an attempt to implement this, I encountered a failing test case (https://github.com/react-tags/react-tags/blob/master/__tests__/reactTags.test.js#L784), which prompted me to reconsider the feasibility of this integration. I would appreciate your insights on this matter

Comment (ad1992) [reply src/components/ReactTags.js:506]: @faizanu94 thats a good point, in that case how about disable adding the tags and show an error message below the input? (the input field will be visible)

Comment (faizanu94) [reply src/components/ReactTags.js]: @ad1992 just added this handling and tried to cover all possible scenarios (clearAll, delete tag, paste etc). Unit test also updated. Awaiting feedback, thanks

File: __tests__/__snapshots__/reactTags.test.js.snap [hunk 5]
```diff
@@ -44,6 +44,7 @@ exports[`Test ReactTags When ClearAll is true should render a clear all button 1
         >
           Clear all
         </button>
+
```
Comment (ad1992) [inline __tests__/__snapshots__/reactTags.test.js.snap:47]:
```suggestion
```

**Review by ad1992 — APPROVED** [review]

**Comment by ad1992** [comment]
Merging, thanks @faizanu94 ✨
